### PR TITLE
Add "New Grid" menu that accepts different sizes.

### PR DESCRIPTION
**Pull request by degustaf**

### CODE REVIEW

File: packages/web/src/svelte/header/NewGrid.svelte [hunk 1]
```diff
@@ -0,0 +1,53 @@
+<script lang="ts">
+    import Modal from "../Modal.svelte";
+    import { boardRepr, solutionToString } from "@sudoku-studio/board-utils";
+    import { setupUserState } from "../../js/user";
+    import { createElement } from "../../js/elements";
+
+    // import { fpuzzlesSizes } from "@sudoku-studio/board-format";
```
Comment (wooferzfg) [inline packages/web/src/svelte/header/NewGrid.svelte:7]: Can this be removed?

Comment (degustaf) [reply packages/web/src/svelte/header/NewGrid.svelte]: Just removed it.

File: packages/board-format/src/f-puzzles.ts [hunk 2]
```diff
@@ -88,7 +88,7 @@ type FPuzzlesQuadruple = {
 
 
 
-const fpuzzlesSizes = {
+export const fpuzzlesSizes = {
```
Comment (wooferzfg) [inline packages/board-format/src/f-puzzles.ts:91]: Could we move this constant to `board-utils` and rename it to something like `gridToBoxSizeMap`? https://github.com/SudokuStudio/SudokuStudio/blob/dev/packages/board-utils/src/board-utils.ts
(Unless you can think of a better place for it?)

**Comment by wooferzfg** [comment]
When I change the grid size and try to select some cells, the position I'm selecting appears to have an incorrect offset. Refreshing the page fixes this issue. Could you look into this bug?

chrome_Z2OTwu4txf

**Comment by degustaf** [comment]
I've figured out why the clicks are going to the wrong place. The click event handler is generated by `getSelectDigitInputHandler`, which takes a grid and caches it.

The solution is probably to make it take a `gridRef`. This will require changing the type of `ElementInfo.getInputHandler` to take a `gridRef`. I wanted to check before actually implementing that change.

**Comment by wooferzfg** [comment]
I think that makes sense, unless there's some way to re-create the event handlers after making a new grid?

**Comment by degustaf** [comment]
Pressing `ctrl`, `alt`, or `shift` is enough to fix the buggy behavior.

But, I can't seem to eliminate the bug. Copying the code from those events into the function that updates the size doesn't seem to work.

**Comment by wooferzfg** [comment]
Would the `gridRef` approach eliminate the bug?

**Comment by wooferzfg** [comment]
I'd suggest going ahead with the implementation of that approach

**Comment by degustaf** [comment]
After digging deeper, I realized that the real problem was that I didn't really grok how Svelte worked. This should fix the clicks going to the wrong cell after generating anew board of a different size.

**Pull request merged**